src/applications/vaos/containers/DateTimeSelectPage.jsx

```diff
@@ -86,47 +43,52 @@ export class DateTimeSelectPage extends React.Component {
     document.title = `${pageTitle} | Veterans Affairs`;
   }
 
-  componentDidUpdate(prevProps, prevState) {
-    if (!prevState.validationError && this.state.validationError?.length > 0) {
+  componentDidUpdate(prevProps) {
+    if (
+      !prevProps.data?.calendarData?.error &&
+      this.props.data?.calendarData?.error?.length > 0
+    ) {
       scrollAndFocus('.usa-input-error-message');
     }
-
-    const prevSelectedSlotsCount =
-      prevProps.data.calendarData?.selectedDates?.length || 0;
-    const newSelectedSlotsCount =
-      this.props.data.calendarData?.selectedDates?.length || 0;
-
-    if (prevSelectedSlotsCount !== newSelectedSlotsCount) {
-      this.validate();
-    }
   }
 
+  getOptionsByDate = selectedDate => {
+    const availableSlots = this.props.availableSlots || [];
```

Comment:
Did we have test coverage of this before? I thought we did, but I could be misremembering.

Reply:
I don't think we had test coverage on it, but added it in this PR.